src/elements/Container/Container.js

```diff
@@ -1,33 +1,68 @@
-import React, { Component, PropTypes } from 'react'
-import classNames from 'classnames'
-import META from '../../utils/Meta'
+import cx from 'classnames'
+import React, { PropTypes } from 'react'
+import META from '../../utils/Meta';
+import { getUnhandledProps, useKeyOnly } from '../../utils/propUtils'
 
 /**
- * A container that gives your content some side padding.
+ * A container limits content to a maximum width
  */
-export default class Container extends Component {
-  static propTypes = {
-    children: PropTypes.node,
-    className: PropTypes.string,
-  }
-
-  static _meta = {
-    library: META.library.semanticUI,
-    name: 'Container',
-    type: META.type.element,
-  }
-
-  render() {
-    const classes = classNames(
-      'sd-container',
-      'ui',
-      this.props.className,
-      'container',
-    )
-    return (
-      <div {...this.props} className={classes}>
-        {this.props.children}
-      </div>
-    )
-  }
+function Container(props) {
+  const {
+    text, left, center, right, justified, fluid,
+    children, className,
+  } = props
+
+  const classes = cx('sd-container ui',
+    useKeyOnly(text, 'text'),
+    useKeyOnly(left, 'left aligned'),
+    useKeyOnly(center, 'center aligned'),
+    useKeyOnly(right, 'right aligned'),
+    useKeyOnly(justified, 'justified'),
```

Comment:
>we'd only require one useValueAndKey mapping rather than 3?

You are correct, I edited my typo shortly after submitting.  See above.

>For example, Button actions.  It feels right that you'd declare a <Button action/> but this breaks down when stipulating that it should be left or right <Button left action/>, but would seem even worse when combining it with an Icon:
>
>`<Button left action right icon/>`

For a prop that may be boolean only _or_ enum, there is the `keyOrValueAndKey()` util which allows us to do this (I assume you meant Input instead of Button):

```js
<Input action />                       // <input class="action" />
<Input action='left' />                // <input class="left action" />
<Input action='right' />               // <input class="right action" />
```

>Or indeed, imposing defined props types with oneOf enums:
>
>`<Button align="left|center|right|justify" />`

I think this is the way to go.  You can see this in use in the [Label for the pointing](https://github.com/TechnologyAdvice/stardust/blob/master/src/elements/Label/Label.js#L142) prop.  A Label can be pointing alone or pointing with a direction, but never more than one.

I think with this one special handling for the justified case we have a pretty good solution:

```js
cx(
  aligned === 'justified' ? 'justified' : useValueAndKey(aligned, 'aligned'),
)
```

If the `aligned` prop is `justified`, then the class will be `justified` only. Otherwise, the class will be the value of the prop and it's key:

```js
<Container aligned='justified' />  // <div class="ui container justified"></div>
<Container aligned='left' />       // <div class="ui container left aligned"></div>
<Container aligned='right' />      // <div class="ui container right aligned"></div>
```